emperor/support_files/js/controller.js

```diff
@@ -431,14 +445,76 @@ define([
    */
   EmperorController.prototype.screenshot = function(type) {
     type = type || 'png';
-    // Render all scenes so it's rendered in same context as save
-    for (var i = 0; i < this.sceneViews.length; i++) {
-      this.sceneViews[i].render();
+
+    if (type === 'png') {
+      // Render all scenes so it's rendered in same context as save
+      for (var i = 0; i < this.sceneViews.length; i++) {
+        this.sceneViews[i].render();
+      }
+      var c = this.renderer.domElement.toDataURL('image/' + type);
+      // Create DOM-less download link and click it to start download
+      var download = $('<a href="' + c + '" download="emperor.' + type + '">');
+      download.get(0).click();
+    } else if (type === 'svg') {
+      // confirm box based on number of samples: better safe than sorry
+      if (this.dm.length >= 9000) {
+        if (confirm('This number of samples could take a long time and in ' +
+           'some computers the browser will crash. If this happens we ' +
+           'suggest to lower the number of segments or use the png ' +
```

Comment:
We no longer have the ability to decrease the number of segments, can you remove that from this message?